src/org/usfirst/frc/team4915/stronghold/commands/MoveStraightPositionModeCommand.java

```diff
@@ -0,0 +1,125 @@
+package org.usfirst.frc.team4915.stronghold.commands;
+
+
+import edu.wpi.first.wpilibj.CANTalon;
+import edu.wpi.first.wpilibj.command.Command;
+import org.usfirst.frc.team4915.stronghold.Robot;
+import org.usfirst.frc.team4915.stronghold.subsystems.DriveTrain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MoveStraightPositionModeCommand extends Command {
+    public static List<CANTalon> motors = Robot.driveTrain.motors;
+    public double inputDistance;
+    public double inputSpeed;
+    private DriveTrain driveTrain = Robot.driveTrain;
+    private List<Double> desiredTicksValue;
+
+    public MoveStraightPositionModeCommand(double inputDistance, double inputSpeed) {
+
+        requires(driveTrain);
+
+        System.out.println("***MoveStraightPositionModeCommand inputDistance: " + inputDistance + "*******");
+        System.out.println("***MoveStraightPositionModeCommand inputSpeed: " + inputSpeed + "*******");
+        
+
+        this.inputDistance = inputDistance;
+        this.inputSpeed = inputSpeed;
+        // Use requires() here to declare subsystem dependencies
+        // eg. requires(chassis);
+    }
+
+    // Called just before this Command runs the first time
+
+    /**
+     * This initializes the variables for the distance calculator.
+     */
+    protected void initialize() {
+        desiredTicksValue = new ArrayList<Double>();
+
+        double ticksToMove = inputDistance * 12 * 1000 / (14 * Math.PI);
+
+        
+
+        for (int i = 0; i < motors.size(); i++) {
+            CANTalon motor = motors.get(i);
+
+            double startingTickValue = motor.getPosition();
+            double endValue = startingTickValue + ticksToMove;
+            if (i >= 2) {
+                // right motors are inverted
+                endValue = startingTickValue - ticksToMove;
+            }
+            System.out.println("!!!!!!!!!" + inputSpeed + "!!!!!!!!!!!");
+           
+
+            desiredTicksValue.add(endValue);
+        }
+    }
+
+    /**
+     * This uses the wheel circumference and the number of rotations to compute
+     * the distance traveled.
+     */
+
+    // Called repeatedly when this Command is scheduled to run
+    protected void execute() {
+
+        System.out.println("+++++++++" + inputSpeed + "++++++++");
+
+        if (inputDistance < 0) {
+            //driveTrain.driveStraight(inputSpeed);
+            driveTrain.driveStraight(.7);
```

Comment:
maybe this `.7` could be moved to a constant field in either this command class or the drivetrain subsystem class